corounit-engine/src/main/kotlin/ru/fix/corounit/engine/ExecutionRunner.kt

```diff
@@ -1,16 +1,29 @@
 package ru.fix.corounit.engine
 
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Semaphore
+import kotlinx.coroutines.sync.withPermit
 
 class ExecutionRunner(private val context: ExecutionContext) {
 
+    private val classesRunsSemaphore = context.configuration.concurrentTestClassesLimit?.let { Semaphore(it) }
+    private val methodsRunsSemaphore = context.configuration.concurrentTestMethodsLimit?.let { Semaphore(it) }
+
     suspend fun executeExecution(execDesc: CorounitExecutionDescriptor) {
         context.notifyListenerAndRunInSupervisorScope(execDesc) {
             for (classDesc in execDesc.children.mapNotNull { it as? CorounitClassDescriptior }) {
                 launch {
-                    ClassRunner(context, classDesc).executeClass()
+                    classesRunsSemaphore.withPermitIfNotNull {
+                        ClassRunner(context, classDesc).executeClass(methodsRunsSemaphore)
+                    }
                 }
             }
         }
     }
+}
+
+suspend fun Semaphore?.withPermitIfNotNull(block: suspend () -> Unit) = if(this != null) {
+    withPermit { block() }
+} else {
+    block()
```

Comment:
this code is needed to avoid any overhead for those who don't want to set up the limits. 

But MAYBE we should add at some limits by default